code/datums/gamemode/objectives/target/locate.dm

```diff
@@ -0,0 +1,49 @@
+/datum/objective/target/locate
+	name = "locate objects"
+	var/list/objects_to_locate = list() //Formatted as String(name) = Path
+
+/datum/objective/target/locate/IsFulfilled()
+	.=..()
+	if(objects_to_locate.len)
+		return FALSE
+	return TRUE
+
+/datum/objective/target/locate/random/find_target()
+	var/list/potential_objects = list("rubber duck" = /obj/item/weapon/bikehorn/rubberducky,
```

Comment:
this could probably stand to gain some possible targets

also, why do we have this as an associative list? we could just pull the names for the items (this doesn't require initializing a new object, you can use `initial()`)